### DISCUSSION TITLE
Using useForm and useFieldArray together

### DISCUSSION THREAD

**Discussion by NickR49** [react-hook-form/react-hook-form#2897] [Q&A] [answered]
Hi,

I'm using useForm and one of the values is an array of parameters. I then use parameters with useFieldArray.
As I add new values to the parameters array the fields object is updated as expected, but the form object only has partial info for the new parameters.

```
  const {
    handleSubmit,
    register,
    reset,
    control,
    errors,
    formState,
    watch,
    setValue,
    getValues,
  } = useForm<ComputedFeedFormData>({
    defaultValues: newComputedFeed,
    mode: 'onBlur',
    reValidateMode: 'onChange',
  });

  const { fields, append, remove } = useFieldArray({
    control,
    name: 'parameters',
    keyName: 'id',
  });
  console.log(`fields: ${JSON.stringify(fields)}`);

  const rawFeed = watch();
  console.log(`rawFeed: ${JSON.stringify(rawFeed)}`);
```

```
fields: [{"id":"new1","scalar":"5"},{"id":"new2","scalar":"22"}]
rawFeed: {"title":"New Computed Feed","description":"Which does awesome things","addConstant":"","operator":"multiply","addRawFeed":null,"parameters":[{"scalar":""},{"scalar":""}]}
```

So the rawFeed form object only has a parameters value of `[{"scalar":""},{"scalar":""}]` instead of `[{"id":"new1","scalar":"5"},{"id":"new2","scalar":"22"}]`. Am I wiring it up wrong?

Here is a code sandbox with the problem - https://codesandbox.io/s/rhf6-usefieldarray-jhcws?file=/src/App.tsx

cheers
Nick

**Comment by bluebill1049** [comment, marked as answer]
you want to watch the specific array field with watch

```
const { fields } = useFieldArray({ control, name: 'fieldArrayName' })
`watch('fieldArrayName',  fields)`
```

**Reply by NickR49** [reply]
Thanks Bill - I've gone with a simpler approach adapted from one of the examples -
https://codesandbox.io/s/react-hook-form-v6-controller-append-prepepend-insert-forked-n8cg7?file=/src/index.js
(though I don't understand why the array needs to be backed by local state instead of the form library itself)

**Reply by bluebill1049** [reply]
hey @NickR49 try to watch only the field array name with `watch`. 

> though I don't understand why the array needs to be backed by local state instead of the form library itself

because `watch` will be invoked before the render, react hook form read value from the input.

let me know if this explains.

**Reply by NickR49** [reply]
Thanks Bill, I did it like you said and it works -
```
  const { register, control, handleSubmit, watch } = useForm();
  const { fields, append, remove } = useFieldArray({
    name: "parameters",
    control
  });
  const parameters = watch("parameters", fields);
```
I got stuck because I hadn't done that and further down I was trying to map over `parameters` instead of `fields`.

**Reply by NickR49** [reply]
BTW, I don't think the API docs for watch explain the overloading for `watch("parameters", fields)`
